phac-nml/irida-next
# Data Export - Listing

## What does this PR do and why?
This PR adds the Data Export listing page. This page will display all the current exports the user has created.

Note: The ID and name of exports will be links that navigate to the data export's respective pages. This will be implemented in a future PR.

## Screenshots or screen recordings
Listing page:
![image](https://github.com/phac-nml/irida-next/assets/82407232/635b60cf-afad-4615-923e-c7800c5e0a78)

Data export in sidebar dropdown:
![image](https://github.com/phac-nml/irida-next/assets/82407232/46a06618-58f2-4a36-939c-e72103ccee48)

Download action for exports with `status = ready`
![image](https://github.com/phac-nml/irida-next/assets/82407232/6880ff51-4b5b-4cc3-aed3-6fe9cbb792f6)

No download action for exports with `status = processing`
![image](https://github.com/phac-nml/irida-next/assets/82407232/7199b5ca-47c5-49e5-807c-9768e18618fa)

Screencast of downloading and deleting export
[Screencast from 2024-03-21 03:27:16 PM.webm](https://github.com/phac-nml/irida-next/assets/82407232/79ab7438-fbbf-44d8-8b47-18e3156ec431)

## How to set up and validate locally
1. In `rails c` `db:drop/create/migrate/seed`. You will now have data exports on users admin and 1 to 7.
2. Ensure navigation to the exports listing page works for both the sidebar link and the sidebar dropdown
3. Once on the exports listing page:
  - Ensure processing is in a gray pill and ready is in a green pill
  - Any exports still processing should have no expiry date
  - Any exports with no name should have the name column blank
  - Only the `status=ready` exports have `download` in the action dropdown
  - Download an export and ensure it downloads properly.
  - Delete exports, ensuring they no longer appear on the listing page after deletion

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
